src/Product.php

```diff
@@ -214,20 +241,66 @@ public static function display_store_stock_status_block() {
     }
 
     global $product;
+    $store_stocks = static::getStockByStore($product);
 
-    $stocks = static::getStockByStore($product);
-    $raw = $stocks;
-    foreach ($stocks as $name => $types) {
+    $raw = $store_stocks;
+
+    foreach ($store_stocks as $name => $types) {
       foreach ($types as $type => $stock) {
-        $stocks[$name][$type] = Stock::renderLevel($stocks[$name][$type], $type);
+        $store_stocks[$name][$type] = Stock::renderLevel($store_stocks[$name][$type], $type);
       }
     }
+
     Plugin::renderTemplate(['templates/store-stock.php'], [
       'raw' => $raw,
-      'stocks' => $stocks,
+      'stocks' => $store_stocks,
+      'product_type' => $product->get_type(),
     ]);
   }
 
+  /**
+   * Returns the initial availability array structure.
+   *
+   * @param array $allVariationsStock
+   *   Stock of all products variations.
+   */
+  public static function buildInitialAvailabilityArray(array $allVariationsStock): array {
+    $availability = [];
+    if (!empty($allVariationsStock) && count($allVariationsStock) > 1) {
+      foreach (reset($allVariationsStock) as $location => $stocks) {
+        $availability[$location] = [];
+        foreach ($stocks as $type => $stock) {
+          if (!isset($availability[$location][$type])) {
+            $availability[$location][$type] = [];
+          }
+        }
+      }
+    }
+    return $availability;
+  }
+
+  /**
+   * Updates the availability with the product ID based on location and stock type.
+   *
+   * @param array $availability
+   *   The avaiability array for all locations.
+   * @param int $product_id
+   *   The ID of the given product.
+   * @param array $locations
+   *   The locations array for the product_id.
+   */
+  public static function getAvailabilityWithProductID(array &$availability, int $product_id, array $locations) {
```

Comment:
`$availability` should not be passed by reference.
```suggestion
  public static function getAvailabilityWithProductID(array $availability, int $product_id, array $locations) {
```